### DISCUSSION TITLE
migrate flow for Angular 13 and nx 13

### DISCUSSION THREAD

**Discussion by Den-dp** [nrwl/nx#7710] [General]
I'm trying to figure out what are the intended steps to migrate `nx 12` workspace with `Angular 12` project to the latest versions (which are 13 for both).

### What I tried:

1. Firstly, as a newcomer, I tried `ng update` and was faced with a suggestion to run `nx migrate latest` (this one is neat).

2. After I run `nx migrate latest` I realized that it only updates nrwl-related packages (still neat but that is half of the work).

3. Then, since [nx states that `nx migrate` is a better version of `ng update`](https://nx.dev/l/a/core-concepts/updating-nx#what-about-ng-update) I tried to adapt what I usually tend to do while updating angular (`ng update @angular/cli @angular/core`) but via nx, so: `nx migrate @angular/cli @angular/core`. But it's turned out that only the first listed package was updated which is definitely not what I want.

4. After that I started running a separate `nx migrate <PackageName>` for each of the packages from my `package.json`
```json
  "dependencies": {
    "@angular/animations": "~12.2.0",
    "@angular/common": "~12.2.0",
    "@angular/compiler": "~12.2.0",
    "@angular/core": "~12.2.0",
    "@angular/forms": "~12.2.0",
    "@angular/platform-browser": "~12.2.0",
    "@angular/platform-browser-dynamic": "~12.2.0",
    "@angular/router": "~12.2.0",
    "rxjs": "~6.6.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4",
    ...
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~12.2.13",
    "@angular/cli": "~12.2.13",
    "@angular/compiler-cli": "~12.2.0",
    "@types/node": "^12.11.1",
    "typescript": "~4.3.5",
    ...
  }
```
but then gave up since to me it feels like I'm missing something and there should be a simpler way to update Angular and its deps.

**Comment by Den-dp** [comment]
Looks like related to https://github.com/nrwl/nx/issues/5648 and https://github.com/nrwl/nx/issues/4575

**Comment by leon** [comment]
nx 13.0.0 is not compatible with angular 13 yet.

It's a lot of work to ensure that nx is fully compatible with angular. 
- cypress
- jest
- eslint

are all tools that are not supported by angular, but by nx.

in angular 13 they changed to a ESM only output.
this caused jest to stop working and jest-preset-angular (not nx), needed to catch up with angular.
jest-preset-angular v11 is going to have support for angular 13, but it's not released yet
https://github.com/thymikee/jest-preset-angular/releases/

also there might be things that need to be migrated.
angular comes with migration scripts for it's things,
and nx 13 came with migration scripts for the workspace.
but now maybe angular 13 in combination with nx 13 needs some migration scripts, which take time to complete.

so when a new version of angular comes along it usually takes a couple of weeks before the ecosystem and the people at nrwl have the time to catch up.

my wild guess is that we are close, since jest-preset-angular is at 11 rc4 and it has been a couple of weeks.

when they release which will probably be nx 13.2 just running `nx migrate latest` will infact update angular.
but there are improvements to be made as @Den-dp pointed out.

**Reply by leon** [reply]
NX 13.2 is now released, so now a `nx migrate latest` should work :)

**Comment by AKrasovski** [comment]
Is NX now officially compatible with Angular 13? The question on google returns this as first answer:
"on Nov 17, 2021. nx 13.0. 0 is not compatible with angular 13 yet. It's a lot of work to ensure that nx is fully compatible with angular."
Since then 4 moths passed - so is it compatible now?
